Finished stream api tasks

myfosse opened the pull request.

Commit:
IEMEREDU13-60: Implement first task about students

Commit:
IEMEREDU13-60: Implement second task about list of tasks

Commit:
IEMEREDU13-60: Implement third task about task converter

Commit:
IEMEREDU13-60:Bug fix after code review

Change student generation function. Split to two difference functions.
Add @UtilityClass annotation to worker classes
Change import to static

ilyaaleks commented on Task1Student/src/main/java/config/StudentWorkerConfig.java:
```diff
@@ -0,0 +1,10 @@
+  public static final int MAX_SUBJECT_MARK = 10;
+  public static final String STUDENT_RANDOM_NAME = "name_";
+  public static final String SUBJECT_RANDOM_NAME = "subject_";
+}
```
Если это класс только с параметрами для конфигурации, то можно аннотировать ```@NoArgsConstructor(access = PRIVATE)```

ilyaaleks commented on Task1Student/src/main/java/worker/StudentWorker.java:
```diff
@@ -0,0 +1,39 @@
+
+    for (int i = 0; i < StudentWorkerConfig.AMOUNT_OF_STUDENTS; i++) {
+      Student student = new Student(StudentWorkerConfig.STUDENT_RANDOM_NAME + i);
+      for (int j = 0; j < StudentWorkerConfig.AMOUNT_OF_SUBJECTS; j++) {
+        student.rate(
+            StudentWorkerConfig.SUBJECT_RANDOM_NAME + j,
+            random.nextInt(StudentWorkerConfig.MAX_SUBJECT_MARK + 1));
+      }
```
Лучше цикл в цикле не писать, а вынести в отдельный метод.
```generateStudentRating(student); ```
например

ilyaaleks replied:
Так же initilCapacity у arraylist не забывай передавать

ilyaaleks replied:
Можно таким способом еще было что-нибудь сгенерировать. Только через peek еще засетать значения
```
final AtomicInteger ordinal = new AtomicInteger(0);
    Stream.generate(()->new Student(StudentWorkerConfig.STUDENT_RANDOM_NAME_PREFIX + ordinal.getAndIncrement())).limit(AMOUNT_OF_STUDENTS).collect(toList());
```

ilyaaleks commented on Task1Student/src/main/java/worker/StudentWorker.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    return students;
+  }
```
```
private void generateStudentRating(final Student student)
  {
    for (int j = 0; j < StudentWorkerConfig.AMOUNT_OF_SUBJECTS; j++) {
      student.rate(
              StudentWorkerConfig.SUBJECT_RANDOM_NAME + j,
              random.nextInt(StudentWorkerConfig.MAX_SUBJECT_MARK + 1));
    }
  }
```

ilyaaleks commented on Task1Student/src/main/java/worker/StudentWorker.java:
```diff
@@ -0,0 +1,39 @@
+
+/** @author Andrei Yahorau */
+public class StudentWorker {
+
```
``` private final Random random = new Random();```

ilyaaleks commented on Task1Student/src/main/java/worker/StudentWorker.java:
```diff
@@ -0,0 +1,39 @@
+package worker;
+
+import config.StudentWorkerConfig;
```
Используй статик импорты, где возможно

ilyaaleks commented on Task1Student/src/main/java/config/StudentWorkerConfig.java:
```diff
@@ -0,0 +1,10 @@
+  public static final int AMOUNT_OF_SUBJECTS = 10;
+  public static final int MAX_SUBJECT_MARK = 10;
+  public static final String STUDENT_RANDOM_NAME = "name_";
+  public static final String SUBJECT_RANDOM_NAME = "subject_";
```
Лучше написать что-то подобное
```
public static final String STUDENT_RANDOM_NAME_PREFIX = "name_";
  public static final String SUBJECT_RANDOM_NAME_PREFIX = "subject_";
 ```

ilyaaleks commented on Task2TaskForRead/src/main/java/worker/TaskWorker.java:
```diff
@@ -0,0 +1,71 @@
+import java.util.stream.Collectors;
+
+/** @author Andrei Yahorau */
+public class TaskWorker {
```
Просмотреть аннотацию @UtilityClass, в этом классе она бы хорошо смотрелась

ilyaaleks commented on Task2TaskForRead/src/main/java/worker/TaskWorker.java:
```diff
@@ -0,0 +1,71 @@
+/** @author Andrei Yahorau */
+public class TaskWorker {
+
+  public String getFirstFiveReadingTasksSortedByCreatedOnTimeToSingleStringByComma(
```
Название метода просто ПУШКА. Евгений Борисов противопоказан к просмотру)

Shotim commented on Task1Student/src/main/java/config/StudentWorkerConfig.java:
```diff
@@ -0,0 +1,14 @@
+
+/** @author Andrei Yahorau */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class StudentWorkerConfig {
```
It looks like this class is used for constants. Why don't we rename it?

Shotim commented on Task1Student/src/main/java/config/StudentWorkerConfig.java:
```diff
@@ -0,0 +1,14 @@
+import lombok.NoArgsConstructor;
+
+/** @author Andrei Yahorau */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
```
static import

Shotim commented on Task2TaskForRead/src/main/java/worker/TaskWorker.java:
```diff
@@ -0,0 +1,73 @@
+    return tasks.stream()
+        .filter(t -> t.getType().equals(TaskType.READING))
+        .limit(5)
+        .sorted(Comparator.comparing(Task::getCreatedOn))
```
static import

Shotim commented on Task2TaskForRead/src/main/java/worker/TaskWorker.java:
```diff
@@ -0,0 +1,73 @@
+            .addTag("books")
+            .addTag("reading");
+
+    return Arrays.asList(task1, task2, task3, task4, task5);
```
static import

Shotim commented on Task3TaskForReadDto/src/main/java/converter/TaskConverter.java:
```diff
@@ -0,0 +1,27 @@
+public class TaskConverter {
+
+  public static List<TaskDto> convertListOfTaskToListOfTaskDto(final List<Task> tasks) {
+    return tasks.stream().map(TaskConverter::convertTaskToTaskDto).collect(Collectors.toList());
```
static import

Commit:
IEMEREDU13-60: Code fix after review by @Shotim

Change class name.
Import all available static methods.

Commit:
IEMEREDU13-60: Add function to find student marks for each subject

Commit:
IEMEREDU13-60: Add function to group tasks by four parameters

Commit:
IEMEREDU13-60: Code fix after command review

Add single stream implementation to find rating for each subject.
Add filtration to check for availability to count average rating for subject.
Add implementation of grouping task by many parameters with grouping class.

Commit:
IEMEREDU13-60(stream task): Add implementation by for and stream
